KidtronicsLanguage/var.cpp

```diff
@@ -182,19 +184,43 @@ var operator+(var lhs, const var& rhs) {
             return newVar;
         }
         case var::VarType::INTEGER:
-            return var(lhs.m_integerValue + rhs.m_integerValue);
+            if(rhs.m_currentType == var::VarType::INTEGER){
+                return var(lhs.m_integerValue + rhs.m_integerValue);
+            }
+            else if(rhs.m_currentType == var::VarType::DOUBLE){
+                return var(lhs.getValueInDouble() + rhs.getValueInDouble());
+            }
+            return lhs;
+        case var::VarType::DOUBLE:
+            return var(lhs.getValueInDouble() + rhs.getValueInDouble());
         default:
             return lhs;
     }
 }
 
+
+
 var operator-(var lhs, const var& rhs) {
-    if (lhs.m_currentType == var::VarType::INTEGER
-        && rhs.m_currentType == var::VarType::INTEGER)
-    {
-        return var(lhs.m_integerValue - rhs.m_integerValue);
+    if (lhs.m_currentType == var::VarType::INTEGER) {
+        if(rhs.m_currentType == var::VarType::INTEGER){
+            return var(lhs.m_integerValue - rhs.m_integerValue);
+        }
+        else if(rhs.m_currentType == var::VarType::DOUBLE){
+            return var(lhs.getValueInDouble() - rhs.getValueInDouble());
+        }
+        else{
+            return lhs;
+        }
     }
-    else {
+    else if(lhs.m_currentType == var::VarType::DOUBLE){
+        if(rhs.m_currentType == var::VarType::INTEGER || rhs.m_currentType == var::VarType::DOUBLE){
+            return var(lhs.getValueInDouble() - rhs.getValueInDouble());
+        }
+        else{
```

Comment:
You can remove this else block and the other following else block. Then have the function return lhs when program reaches the end of the function.